python/ray/dag/compiled_dag_node.py

```diff
@@ -1431,7 +1431,9 @@ async def execute_async(
             fut = asyncio.Future()
             await self._fut_queue.put(fut)
 
-        return CompiledDAGFuture(self, self._execution_index, fut)
+        fut = CompiledDAGFuture(self, self._execution_index, fut)
+        self._execution_index += 1
```

Comment:
is it related to this PR?

Reply:
Just a fix for UX - the execution index here is only used for the `__str__` representation of the future, but it always said execution_index=0 before :)